### PR TITLE
Improve test coverage for store#findRecord()

### PR DESCRIPTION
**Pull request by HeroicEric**
Started working on fixing https://github.com/emberjs/data/issues/3704 by trying to write a failing test but was not able to reproduce the issue.

Am I missing something?

### CODE REVIEW

**Comment by wecc** [comment]
LGTM

**Comment by HeroicEric** [comment]
@irnc am I misunderstanding the bug you were running into?

File: packages/ember-data/tests/integration/store-test.js [hunk 1]
```diff
@@ -254,22 +253,58 @@ test("Using store#findRecord on existing record reloads it", function() {
...
+    });
+  });
+
+  run(function() {
+    let car = store.peekRecord('car', 1);
+    equal(car.get('model'), 'Princess', 'car record was reloaded');
+  });
+});
+
+test("store#findRecord { reload: true } ignores cached record and reloads record from server", function() {
```
Comment (bmac) [inline packages/ember-data/tests/integration/store-test.js]: @HeroicEric in this test could you also assert that `shouldReloadRecord` is never called when `{ reload: true }` is passed into findRecord?

**Comment by bmac** [comment]
Thanks @HeroicEric this looks great!

File: packages/ember-data/tests/integration/store-test.js (hunk 1 above)
Comment (HeroicEric) [inline packages/ember-data/tests/integration/store-test.js]: added assertion at https://github.com/emberjs/data/pull/3746/files#diff-a40f0b2297a7d2fb4022eb1fcae21af9R280

**Comment by bmac** [comment]
Thanks @HeroicEric

**Comment by irnc** [comment]
@HeroicEric Tests look good. I should have misread code, so my report is incorrect, based on how I read it right now.

But there is definitely an issue for my app, which I can reproduce, but can not debug it right now because Chrome does not stop on breakpoint in `store.js` (which is server in ES6 from source maps for some reason).

**Comment by bmac** [comment]
@irnc try disabling sourcemaps in your chrome dev tools.

screen shot 2015-09-11 at 2 37 39 pm

**Pull request merged**